### PR TITLE
Add integration tests on a self-hosted runner

### PR DESCRIPTION
**Pull request by lebrice**
- Adds a self-hosted github runner for integration tests (with ssh access to real slurm clusters!)
- Updates a few tests so they also run correctly on a self-hosted CI runner (with real access to the SLURM clusters).
- Adds the `pytest-skip-slow` plugin to skip "integration" tests: they are now simply marked with `@pytest.mark.slow`.

### CODE REVIEW

File: tests/cli/test_local.py [hunk 1]
```diff
@@ -126,40 +132,65 @@ def paramiko_openssh_key_parsing_issue(strict: bool = False):
     )
 
 
+# TODO: This test is incredibly flaky. Needs to be totally reworked.
+@PARAMIKO_SSH_BANNER_BUG
```
Comment (satyaog) [inline tests/cli/test_local.py:136]: `@PARAMIKO_SSH_BANNER_BUG` is applied to the test but the code below seams to also specify which tests are expected to fail. Why is that?

File: tests/utils/test_parallel_progress.py [hunk 2]
```diff
@@ -112,5 +112,6 @@ def assert_increasing(sequence: Sequence[float]) -> None:
     # longest task was programmed to take a known amount of time to run, so the
     # overall progress bar should have taken a max of ~ `max(task_lengths)` seconds.
     longtest_task_length = max(task_lengths)
-    # note: somewhat arbitrary offset here, was sometimes failing in the CI with just 1.
-    assert longtest_task_length <= total_time_seconds <= longtest_task_length + 1.5
+    # NOTE: This check was flaky. Adding some slack to the max time only temporarily
+    # addressed this.
+    assert longtest_task_length <= total_time_seconds  # <= longtest_task_length + 1.5
```
Comment (satyaog) [inline tests/utils/test_parallel_progress.py:117]: If we want to test if we save time by doing parallel tasks maybe we could run some parallel `sleep 10` and make sure the total time is not larger than X times 10s?

**Review by satyaog — APPROVED** [review]

File: tests/cli/test_local.py (hunk 1 above)
Comment (lebrice) [reply tests/cli/test_local.py:136]: Yeah I got lazy and just added it to all cases. It's a mess. There's apparently no way to guaranteed that it's not going to happen. Fixed in bc473f890cd570a279cd29ad71e5bb2e5cee0e7f

File: tests/utils/test_parallel_progress.py (hunk 2 above)
Comment (lebrice) [reply tests/utils/test_parallel_progress.py]: Fixed in b035f0610cde46be0a56004d4a39979f3c388391

**Review by satyaog — APPROVED** [review]

**Pull request merged**